src/parse.c

```diff
@@ -1543,6 +1547,14 @@ handle_optional_addresses(NetplanParser* npp, yaml_node_t* node, const void* _,
 
         for (unsigned i = 0; NETPLAN_OPTIONAL_ADDRESS_TYPES[i].name != NULL; ++i) {
             if (g_ascii_strcasecmp(scalar(entry), NETPLAN_OPTIONAL_ADDRESS_TYPES[i].name) == 0) {
+
+                /* Values below NETPLAN_OPTIONAL_STATIC (including it) are not valid and
+                 * considered deprecated. See LP: #1880029
+                 */
+                if (NETPLAN_OPTIONAL_ADDRESS_TYPES[i].flag <= NETPLAN_OPTIONAL_STATIC)
+                    g_warning("Flag \"%s\" in optional-addresses is deprecated. Valid values are: "
```

Comment:
We shouldn't deprecate the original keywords, but rather try to get them implemented in the corresponding backend renderers.

That is especially:
* Waiting LL configuration only
  * This can already be achieved on networkd, by using `RequiredForOnline=degraded` (instead of `routable/yes`). Can be in combination with `RequiredFamilyForOnline=ipv4/ipv6`. – But what happens if we want to have static or dynamic IP4 but only LL IP6?
* Waiting for static IP configuration, but not dynamic/DHCP.